test/search/query-variants.test.ts

```diff
@@ -0,0 +1,48 @@
+import {queryVariants} from '../../src/search/query-variants';
+
+describe('Search query', () => {
+  it('transforms simple query', () => {
+    expect(queryVariants('test')).toEqual(
+      new Map([
+        ['?query', 'test'],
+        ['?booleanQuery', "'test'"],
+      ])
+    );
+  });
+
+  it('transforms multiple word query', () => {
+    expect(queryVariants('Dr. H. Colijnstraat')).toEqual(
```

Comment:
A rather nasty one is this search word: `Sammy Dowds & Leslie` (Muziekweb). This should result in `'Sammy' AND 'Dowds &' AND 'Leslie'`, not in `'Sammy' AND 'Dowds' AND '&' AND 'Leslie'` (Virtuoso will complain about the `'&'`: `phrase consists of noise words exclusively`). Alternatively, we could remove characters (or sequences of characters) that only consists of 'noise words' (e.g. `&` or `&&`).

Reply:
The latter makes most sense. However, it seems [the set of noise words is configurable](http://docs.openlinksw.com/virtuoso/noisewords/), so how can we know which noise words to remove?

Reply:
Ouch, I wasn't aware of the configurability of the noise words. So we cannot with confidence remove specific characters, such as `&`; a Virtuoso instance may have configured other noise words.

Could we continue with the first option, by not removing the noise word but by including it in the quoted part of a search word (e.g. `Dowds &`)? This is not a perfect solution, though: if a Virtuoso instance has configured `van` to be a noise word, and the search word of a NoT user is `van gogh`, our query pre-processor wil create the query `'van' AND 'gogh'`, resulting in the Virtuoso error `phrase consists of noise words exclusively`.

Reply:
I think we still need a solution for this one: `Sammy Dowds & Leslie` - right?

`Dowds &` and `Dr.` are basically the same to Virtuoso: they both contain noise words, `&` and `.`, that are only allowed as part of a phrase that includes non-noise words. The difference, however, is that our `SMART` pre-processor shouldn't split the search words of a user on whitespace. Rather, it should keep non-noise (`Dowds`) and noise (` &`) together. Is this possible?

Reply:
Changed this to just remove stop words. While this loses some precision in the query, which drops the ampersand completely and becomes `'sammy' AND 'dowds' AND 'Leslie'`, this makes the pre-processing easier to maintain for us.